app/controllers/automated_tests_controller.rb

```diff
@@ -193,12 +199,13 @@ def upload_specs
     if params[:specs_file].respond_to? :read
       file_content = params[:specs_file].read
       begin
-        JSON.parse file_content
+        test_specs = JSON.parse file_content
       rescue JSON::ParserError
         flash_now(:error, I18n.t('automated_tests.invalid_specs_file'))
         head :unprocessable_entity
       else
         File.write(assignment.autotest_settings_file, file_content, mode: 'wb')
+        update_test_groups_from_specs(assignment, test_specs)
```

Comment:
I like reusing this method, but given this call there's now more opportunity to have a runtime error (I just encountered one when uploading an old specs file that has an int instead of string for the criterion field).

How about we use a begin-rescue block around this call and flash an error message if an exception gets raised. Additionally, I think the file should only be saved and sent to the autotesting server if there's no exception?